plugins/kotlin/idea/src/org/jetbrains/kotlin/idea/codeInliner/CodeToInlineBuilder.kt

```diff
@@ -357,8 +358,10 @@ class CodeToInlineBuilder(
                     if (receiver is ImplicitReceiver) {
                         val resolutionScope = expression.getResolutionScope(bindingContext, resolutionFacade)
                         val receiverExpressionToInline = receiver.asExpression(resolutionScope, psiFactory)
-                        if (receiverExpressionToInline != null) {
-                            val receiverType = receiver.type.unCapture()
+                        val receiverType = receiver.type.unCapture()
+                        val receiverIsUnnecessary = (receiverExpressionToInline as? KtThisExpression)?.labelQualifier != null &&
+                                receiverType == targetCallable.receiverType()
```

Comment:
Shouldn't we do `unCapture()` for `targetCallable.receiverType()` too?

Reply:
b2a478825b17